route/kafkamdm.go

```diff
@@ -124,18 +125,38 @@ func NewKafkaMdm(key, prefix, sub, regex, topic, codec, schemasFile, partitionBy
 func (r *KafkaMdm) run() {
 	metrics := make([]*schema.MetricData, 0, r.flushMaxNum)
 	ticker := time.NewTicker(r.flushMaxWait)
+	var client sarama.Client
 	var err error
 
 	for r.producer == nil {
-		r.producer, err = sarama.NewSyncProducer(r.brokers, r.saramaCfg)
+		client, err = sarama.NewClient(r.brokers, r.saramaCfg)
 		if err == sarama.ErrOutOfBrokers {
 			log.Warnf("kafkaMdm %q: %s", r.key, err)
 			// sleep before trying to connect again.
 			time.Sleep(time.Second)
 		} else if err != nil {
 			log.Fatalf("kafkaMdm %q: failed to initialize kafka producer. %s", r.key, err)
 		}
+
+		partitions, err := client.Partitions(r.topic)
+		if err != nil {
+			log.Fatalf("kafkaMdm %q: failed to get partitions for topic %s - %s", r.key, r.topic, err)
+		}
+		if len(partitions) < 1 {
+			log.Fatalf("kafkaMdm %q: failed to get partitions for topic %s", r.key, r.topic)
```

Comment:
this error seems wrong. it should tell the user that the partitions were obtained, but there were `0` partitions